### DISCUSSION TITLE
App/Libraries Licensing

### DISCUSSION THREAD

**Discussion by thebino** [AndroidDev-social/DodoForMastodon#23] [General]
### Description

There are many licenses out there, we should find a license where most of the contributors are happy with.

- should not prohibit the usage of closed source libraries
- should prevent people from making money based on the community contribution

### Additional Information

- https://opensource.org/licenses
- https://creativecommons.org/about/cclicenses/

![Screenshot 2022-11-08 at 15 59 23](https://user-images.githubusercontent.com/841681/200598643-9b53795d-5d3c-443c-8c49-9dbf482a3c98.png)

**Comment by evant** [comment]
How about MPL? It requires redistribution of  modified source files under MPL while also allowing linking to closed-source deps
 https://www.mozilla.org/en-US/MPL/2.0/FAQ/

**Comment by digitalbuddha** [comment]
Any reason to not go with apache or MIT?

**Reply by blvckcoded** [reply]
Yes, at least with Apache someone could clone this repo, change naming and branding and then sell this. We will want a license that prevents anyone from profiting from our work.

**Reply by digitalbuddha** [reply]
My two cents

I'm a proponent of open source allowing folks to do what they please. If someone wants to package and sell they should be able to just a much as I would be able to package/sell work I made with a square lib, rxjava or others. 

I don't believe the goal should be to keep others from profiting off of this, maybe someone will be able to add some premium features/package it for a particular server and sell it. I'm fine with that personally.

My initial goal was to have a client that folks from Androiddev.social would be able to freely contribute to/update etc. If others feel differently feel free to proceed with a more strict license.

**Reply by blvckcoded** [reply]
FWIW - I'm not opposed to that either. My passion behind this was mainly working with bleeding edge tech, getting to know the community and building something open source that is challenging and honestly fun/interesting. We will do a poll alongside other decisions that need to be made so we can really slam the gas pedal and take off work starting the next few days. 

All good points you raised 🙌

**Comment by evant** [comment]
There seems to be a good amount of copyleft in this space, seems like a reasonable thing to support imo.

**Comment by blvckcoded** [comment]
Based on this slack thread: I think we should go with GPL license

https://androiddev-social.slack.com/archives/C04A9U5CBSM/p1668015004906239?thread_ts=1668015004.906239&cid=C04A9U5CBSM

**Comment by exallium** [comment]
My opinion is that there are two component here, the libraries and the end user product.

I think that the libraries we produce (the things interacting with Mastodon API in a KMP manner, etc) can all be unrestrictively open sourced, using a licence like Apache 2, whereas the end product that is built, the actual deliverable that end users download and use, should be more restrictive. Some kind of copy-left license. We don't want a situation where we're chasing around people creating clones and trying to make a quick buck off of them, but we still want to empower devs to be able to build off of what is created here.

**Comment by digitalbuddha** [comment]
That sounds like a good compromise. I'd love to be able to reuse most of the modules, create my own ui and then package as my own. Would that be something that would be fine?

**Reply by exallium** [reply]
This sounds fine to me. We want to strike a balance between enabling flips and empowering folk to build their own clients as simply as possible.

**Comment by blvckcoded** [comment]
I would be down with even pulling the UI into its own library, and stripping out references to our theme and having that be set on the consumer/app side.

This allows the community to build upon the component library or just use it too. It should have a lot of good stuff in there as well. (Maybe elegant animations etc)

**Comment by thebino** [comment]
⛔️ This discussion is closed. Please vote or [check the poll for results.](https://github.com/AndroidDev-social/MastodonCompose/discussions/53)
